pulp_smash/utils.py

```diff
@@ -80,3 +82,38 @@ def poll_task(server_config, href):
         # This approach is dumb, in that we don't account for time spent
         # waiting for the Pulp server to respond to us.
         sleep(5)
+
+
+def get_broker(server_config):
+    """Build an object for managing the target system's AMQP broker.
+
+    Talk to the host named by ``server_config`` and use simple heuristics to
+    determine which AMQP broker is installed. If Qpid or RabbitMQ appear to be
+    installed, return a :class:`pulp_smash.cli.Service` object for managing
+    those services respectively. Otherwise, raise an exception.
+
+    Another option for solving this problem is to create a subclass of
+    ``Service``, like so:
+
+    >>> class Broker(Service):
+    ...     def __init__(self, server_config):
+    ...         service = …  # 'qpidd', 'rabbitmq', etc.
+    ...         super(Broker, self).__init__(server_config, service)
+
+    However, this approach has two drawbacks:
+
+    * It favors inheritance over composition.
+    * It violates the Liskov substitution principle, due to its ``__init__``
+      method requiring a more specific set of arguments.
+
+    :param pulp_smash.config.ServerConfig server_config: Information about the
+        system on which an AMQP broker exists.
+    :rtype: pulp_smash.cli.Service
+    """
+    client = cli.Client(server_config, cli.echo_handler)
+    for executable in ('qpidd', 'rabbitmq'):  # ordering is important
+        file_check = ('[', '-f', '/usr/sbin/' + executable, ']')
```

Comment:
`/etc/profile` is only executed for login shells. Fedora 23 adds `/usr/sbin` and `/usr/local/sbin` to `$PATH` in `/etc/profile`. As a result, `ssh fedora-23-host.example.com which qpidd` returns a non-zero exit code.